tests/test_encounter/test_window_generator.py

```diff
@@ -0,0 +1,142 @@
+
+
+from celest.coordinates.coordinate import Coordinate
+from celest.coordinates.frames.azel import AzEl
+from celest.coordinates.frames.gcrs import GCRS
+from celest.coordinates.ground_location import GroundLocation
+from celest.encounter.window_generator import (
+    generate_vtw,
+    _get_day_constraint_indices,
+    _get_night_constraint_indices,
+    Lighting,
+    _sun_coordinates
+)
+from celest.satellite import Satellite
+from celest import units as u
+from unittest import TestCase
+import numpy as np
+
+
+class TestGenerateVTW(TestCase):
+
+    def setUp(self):
+        data = np.loadtxt(
+            fname="tests/test_data/coordinate_validation_long.txt",
+            usecols=(0, 11, 12, 13, 14, 15, 16),
+            skiprows=1,
+            max_rows=5000
+        )
+
+        self.times = data[:, 0] + 2430000
+        self.gcrs = data[:, 1:4]
+        self.gcrs_velocity = data[:, 4:]
```

Comment:
Change variable name since we override it or change `self.gcrs` to `self.gcrs_velocity`. Maybe do the former.